### PR TITLE
Address the `Entity` model in `Arbitrary[Entity[*]]`

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

File: laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala [hunk 1]
```diff
@@ -860,7 +860,15 @@ private[discipline] trait ArbitraryInstances { this: ArbitraryInstancesBinCompat
...
-      } yield Entity(body, length)
+        entity = length match {
+          case Some(0) =>
+            Entity.Empty
+          case Some(_) =>
+            Entity.strict(body.compile.to(fs2.Chunk))
+          case None =>
+            Entity(body)
+        }
+      } yield entity
```
Comment (armanbilge) [inline laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala:871]: What do you think about having separate generators for the different types of entities, and then choosing between them?

Comment (danicheg) [reply laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala:871]: In fact, they'll serve only one generator and won't be reused. So it's a question about tastes, I think.

Comment (armanbilge) [reply laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala:871]: I think it also helps for reasoning about the generator. For example, this generator doesn't included a chunked streaming entity with a known length, but I didn't realize it until now.

Comment (danicheg) [reply laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala]: Touche 🤔 Applied your advice in a recent patch

**Review by armanbilge — APPROVED** [review]
Looks good, thanks :)

**Pull request merged**